### PR TITLE
Data population using populate() instead of lean() in acceptMembershipRequest.ts

### LINKED ISSUE #1270: Use populate() instead of lean() when querying for the Organization and User documents in acceptMembershipRequest.ts
**Describe the bug**
 Using populate() will allow us to get the referenced documents along with the current document in a single query instead of multiple queries. This can significantly reduce the number of database queries made, leading to better performance.

**To Reproduce**

1. start the local server using 'npm run dev'
2. run the query to acceptMembershipRequest

**Expected behavior**
Using populate() will reduce the multiple queries to the database leading to a better performance

**Actual behavior**
The current implementation use .lean() to fetch the documents independently with multiple queries, which can result in slow response.

**Screenshots**
image

**Additional details**
Not required

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by nishant3721**
**What kind of change does this PR introduce?**

Refactoring

**Issue Number:**

Fixes #1270 

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

image

**If relevant, did you update the documentation?**

Not relevant

**Does this PR introduce a breaking change?**

No

**Other information**

Not required

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

File: src/resolvers/Mutation/acceptMembershipRequest.ts [hunk 1]
```diff
@@ -65,7 +68,7 @@ export const acceptMembershipRequest: MutationResolvers["acceptMembershipRequest
     await adminCheck(context.userId, organization);
 
     const userIsOrganizationMember = organization.members.some(
-      (member) => member.toString() === user?._id.toString()
+      (member: string) => member.toString() === user?._id.toString()
```
Comment (EshaanAgg) [inline src/resolvers/Mutation/acceptMembershipRequest.ts:71]: Wrong type annotation. Member wouldn't be a string, but a Mongoose Object ID (That's why you need to use the `.toString()` method)
Please fix the same.

File: src/resolvers/Mutation/removeOrganization.ts [hunk 2]
```diff
@@ -54,124 +54,52 @@ export const removeOrganization: MutationResolvers["removeOrganization"] =
...
-      await Post.deleteOne({
-        _id: postId,
-      });
-
-      await Comment.deleteMany({
-        post: postId,
-      });
-    });
+    await Post.deleteMany({ _id: { $in: organization.posts } });
+    await Comment.deleteMany({ post: { $in: organization.posts } });
```
Comment (EshaanAgg) [inline src/resolvers/Mutation/removeOrganization.ts:58]: All of these changes are from your other PR. Remove them from here.

File: tests/resolvers/Mutation/acceptMembershipRequest.spec.ts [hunk 3]
```diff
@@ -267,13 +267,7 @@ describe("resolvers -> Mutation -> acceptMembershipRequest", () => {
...
     const { acceptMembershipRequest: acceptMembershipRequestResolver } =
       await import("../../../src/resolvers/Mutation/acceptMembershipRequest");
-    const acceptMembershipRequestPayload =
-      await acceptMembershipRequestResolver?.({}, args, context);
-
-    expect(acceptMembershipRequestPayload).toEqual(
-      testMembershipRequest!.toObject()
-    );
-
+    await acceptMembershipRequestResolver?.({}, args, context);
```
Comment (EshaanAgg) [inline tests/resolvers/Mutation/acceptMembershipRequest.spec.ts:270]: Why have you removed this `expect()` assertion?

**Review by EshaanAgg — CHANGES_REQUESTED** [review]

File: src/resolvers/Mutation/acceptMembershipRequest.ts (hunk 1 above)
Comment (nishant3721) [reply src/resolvers/Mutation/acceptMembershipRequest.ts:71]: image
it is giving error, if we doesn't define the type to it.

File: tests/resolvers/Mutation/acceptMembershipRequest.spec.ts (hunk 3 above)
Comment (nishant3721) [reply tests/resolvers/Mutation/acceptMembershipRequest.spec.ts]: This was giving an error after implementing populate(). Issue solved now!

File: src/resolvers/Mutation/acceptMembershipRequest.ts [hunk 4]
```diff
@@ -94,19 +100,18 @@ export const acceptMembershipRequest: MutationResolvers["acceptMembershipRequest
...
         },
       }
     );
 
-    /*
-   Add organization._id to joinedOrganizations list and remove 
-   membershipRequest._id from membershipRequests list on user's document.
-   */
+    /* This code is updating the `User` document in the database by adding the `organization._id` to
+    the `joinedOrganizations` array and removing the `membershipRequest._id` from the
```
Comment (xoldyckk) [inline src/resolvers/Mutation/acceptMembershipRequest.ts:110]: These comments are useless, any person using mongoose should have an idea of what's being done. Even if you're doing it these comments can still be trimmed to a considerable amount.

Comment (nishant3721) [reply src/resolvers/Mutation/acceptMembershipRequest.ts:110]: Okay sir

**Review by xoldyckk — COMMENTED** [review]
Btw the populate method sends seperate queries in the background to fetch the new objects, so this isn't really decreasing the number of queries to database, it's just making the code cleaner.

**Review by xoldyckk — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

**Pull request merged**